integration_tests/pages/page.ts

```diff
@@ -50,12 +50,12 @@ export default abstract class Page {
   }
 
   hasErrorCount(expected: number) {
-    cy.get('.govuk-error-summary__list').should('have.length', expected)
+    cy.get('.govuk-error-summary__list li').should('have.length', expected)
     return this
   }
 
   hasFieldInError(field: string) {
-    cy.get(`#${field}-error`).should('exist')
+    cy.get(`#${Cypress.$.escapeSelector(field)}-error`).should('exist')
```

Comment:
We need to use the cypress method to escape the selectors, because the `id`s of the elements that our new test are looking for contain "special" characters (the `[` and `]`)
EG:
```
    createGoalPage //
      .hasErrorCount(2)
      .hasFieldInError('goals[0].title')
      .hasFieldInError('goals[0].steps[0].title')
```
in the cypress test above ☝️